presto-main/src/main/java/com/facebook/presto/memory/MemoryPool.java

```diff
@@ -351,6 +353,25 @@ private synchronized void updateTaggedMemoryAllocations(QueryId queryId, String
     @VisibleForTesting
     synchronized Map<QueryId, Map<String, Long>> getTaggedMemoryAllocations()
     {
-        return ImmutableMap.copyOf(taggedMemoryAllocations);
+        Map<QueryId, Map<String, Long>> queryIdMap = new HashMap<>();
+        taggedMemoryAllocations.forEach(
+                (queryId, memoryAllocations) -> {
+                    queryIdMap.put(
+                            queryId,
+                            getTaggedMemoryAllocationsByID(queryId));
+                });
+        return ImmutableMap.copyOf(queryIdMap);
+    }
+
+    @VisibleForTesting
+    synchronized Map<String, Long> getTaggedMemoryAllocationsByID(QueryId targetQueryId)
```

Comment:
Sorry, missed this. You can just name this function `getTaggedMemoryAllocations`. The signature is obviously suggesting it's "by ID" so it's not necessary.

Reply:
> getTaggedMemoryAllocations

but we have another getTaggedMemoryAllocations which is called in the test, so I decided to preserve that function and created this new function.
how about changing to getTaggedMemoryAllocation  (without the s) ?

Reply:
That's fine. You can just overload the function name.

Reply:
> That's fine. You can just overload the function name.

I see your point, Done and Done.